blip.go

```diff
@@ -65,6 +65,10 @@ type MetricValue struct {
 
 	// Meta is optional key-value pairs that annotate or describe the metric value.
 	Meta map[string]string
+
+	// The frequency that this metric is collected at. Some sinks need this value
+	// to properly process COUNTER type metrics.
+	Frequency int
```

Comment:
There's a better, cleaner way to achieve this: add `Frequency` to `blip.Metrics` struct then set it at https://github.com/cashapp/blip/blob/main/monitor/engine.go#L273-L279

Then you can remove `collectorFrequencyWrapper` and everything related.

Reply:
Where we getting the `Frequency` value to set it there? We could pass it through as an arg to `engine.Collect` but the level info we end up with from [level_collector.go](https://github.com/cashapp/blip/blob/main/monitor/level_collector.go#L172-L177) is not necessarily the frequency that the collector is running at. For example, we have collectors that run every 30 seconds and others that run every 5 minutes. When the 5 minute collectors run the 30 second collectors also run, but the level that is pulled in `level_collector.go` is the index for the 5 minute collector, which is wrong for the 30 second collectors.

Reply:
What about updating `atLevel` in the `Engine` [struct](https://github.com/cashapp/blip/blob/main/monitor/engine.go#L37-L58) to contain a wrapper type instead of just `Collector`? We could attach the frequency there and then use it at the suggested location without needing `collectorFrequencyWrapper`.

Reply:
I take that back. I don't think that `Frequency` can live on `blip.Metrics` as that houses `MetricValue`s from multiple collectors that could potentially have different frequencies associated with them.

Reply:
Hm, yes, metrics "level up" in overlapping frequencies, so a metric collected every 10s is collected every 60s, too.

Are you certain Datadog needs a frequency? I've seen/used many metrics systems, but I've never seen one require a frequency to calculate counters. (I'm kind of new to Datadog, though.) It doesn't make sense because it's possible and common to calculate counters (or rates from counters) without knowing the configured collection frequency.

Reply:
I played around some more with this in the UI by removing the interval setting on data. The raw metrics for `COUNT` types still display, but any rate calculations fail to display at higher resolutions (appears to be sub-hour?). Putting the interval back fixes the issue. It looks like the interval dictates how DD displays things. If I set the interval to something more frequent than how often we collect the data points then the same issue occurs. I can set it more course (e.g. 60 seconds rather than 30) at which point the resolution of the rate calculations matches the interval, meaning it displays a data point every 60 seconds when zoomed in.

Reply:
_Technically_ we can set this interval (and other metadata) in the UI without sending it over the UI. It means manually adjusting these values for each new metric we bring online. The interval data is shared across all of the metrics with the same name, so if we had a conflict with how the frequency was configured across blip instances that would result in the interval value ping-ponging between different values, which would be less than ideal. Part of me thinks that having the interval, units, etc. get set by the sink based on the configuration would be convenient but it's not a show stopper.